app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -118,6 +118,24 @@ def authenticate_client
       end
     end
 
+    def valid_auth_token?(token)
+      hashed_valid_auth_tokens.any? do |hashed_valid_token|
+        cost = hashed_valid_token[:cost]
+        salt = hashed_valid_token[:salt]
+        hashed_token = scrypt_digest(token: token, salt: salt, cost: cost)
```

Comment:
they all have the same cost/salt right? so we *could* scrypt the incoming value once?

Maybe inside the processing block, we reorganize the secret value so it's more like a dictionary? or even a struct?

```
{
  cost: cost,
  salt: salt,
  hashed_tokens: ['a', 'b', 'c']
}
```

Reply:
yep, that makes sense.